File: data_structures_java/MyStack.java
class MyStack<T> {

	int max;
	T[] arr;
	int top;

	MyStack() {
		this(10);
	}
	MyStack(int max) {
		this.max = max;
		// arr = new T[max];
		arr = (T[]) new Object[max];
		top = -1;
	}
	boolean isEmpty() {
		if (top == -1) {
			return true;
		}
		return false;
	}
	boolean isFull() {
		if (top == max-1) {
			return true;
		}
		return false;
	}
	void push(T item) {
		if (!isFull()) {
			arr[++top] = item;
		}
	}
	T pop() {
		if (!isEmpty()) {
			return arr[top--];
		}
		return arr[0];
	}
	void printStack() {
		for (int i=0; i<=top; i++) {
			System.out.println(arr[i]);
		}
	}

	public static void main(String[] args){
		MyStack<String> ms = new MyStack<String>(5);
		System.out.println(ms.isEmpty());
		System.out.println(ms.isFull());

		ms.push("10fdfsdfs");
		ms.push("5fdfsdfs");
		ms.push("30fdfsdfs");
		ms.printStack();
		
		ms.pop();
		ms.pop();
		ms.pop();
		ms.printStack();
	}
}

File: data_structures_py/mystack.py
class MyStack:

	def __init__(self, max=10):
		self.arr = []
		self.top = -1
		self.max = max

	def isEmpty(self):
		if self.top == -1:
			return True;
		return False;

	def isFull(self):
		if self.top == self.max-1:
			return True
		return False

	def push(self, item):
		if not self.isFull():
			self.top += 1
			self.arr.append(item)
	
	def pop(self):
		if not self.isEmpty():
			to_return = self.arr[self.top]
			del self.arr[self.top]
			self.top -= 1
			return to_return

	def printStack(self):
		for i in self.arr:
			print(i)


if __name__ == "__main__":
	ms = MyStack()

	ms.push(10)
	ms.push(14)
	ms.push(11)
	ms.printStack()

	ms.pop()
	ms.pop()
	ms.push(110)
	ms.printStack()
